zerver/tests/test_sessions.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+
+from zerver.lib.sessions import (
+    user_sessions,
+    delete_session,
+    delete_user_sessions,
+    delete_realm_user_sessions,
+    delete_all_user_sessions,
+    delete_all_deactivated_user_sessions,
+)
+
+from zerver.models import (
+    UserProfile, get_user_profile_by_id,
+    get_user_profile_by_email, get_realm
+)
+
+from zerver.lib.test_classes import ZulipTestCase
+
+
+class TestSessions(ZulipTestCase):
+    def test_delete_session(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        for session in user_sessions(user_profile):
+            delete_session(session)
+        result = self.client_get("/")
+        self.assertEqual('/login', result.url)
+
+    def test_delete_user_session(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        delete_user_sessions(user_profile)
+        result = self.client_get("/")
+        self.assertEqual('/login', result.url)
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_user_sessions(user_profile)
+        self.assertIn('_auth_user_id', self.client.session)
+
+    def test_delete_realm_user_sessions(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        realm = get_realm('zulip')
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_realm_user_sessions(realm)
+        result = self.client_get('/')
+        self.assertEqual('/login', result.url)
+
+    def test_delete_all_user_sessions(self):
+        self.login('hamlet@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        self.login('othello@zulip.com')
+        self.assertIn('_auth_user_id', self.client.session)
+        delete_all_user_sessions()
+        result = self.client_get('/')
+        self.assertEqual('/login', result.url)
```

Comment:
Look at `do_test`  in `zerver/lib/test_events.py` for an example of this sort of thing.  But basically we're using a lambda here just to allow deferring the call to `delete_all_user_sessions()` so that it can be called at the right time inside `do_test_session` (namely, after it's logged the user in), rather than before `do_test_session` .

I imagine Google can teach you a bit about how lambdas work as well :)